web/themes/custom/itkdev/itkdev_project_theme/js/charts.js

```diff
@@ -0,0 +1,90 @@
+import Chart from '../node_modules/chart.js/auto'
+
+(function (Drupal, once, drupalSettings) {
+
+  'use strict';
+
+  Drupal.behaviors.itkProjectThemeBehavior = {
+    attach: function (context, settings) {
+      const chart = once('itkProjectThemeBehavior', document.getElementById('chart'));
+      const chartSettings = drupalSettings['reports_project_track'];
+      let datasets = [];
+
+      if (chart) {
+        chartSettings['dataset'].forEach(function(drupalData) {
+          datasets.push({
+              label: drupalData.chart.label,
+              title: drupalData.chart.label,
+              data: drupalData.plots,
+              backgroundColor: drupalData.chart.color
+            });
+        });
```

Comment:
If I understand this code correctly, you're mapping `chartSettings['dataset']` to `datasets`:

```suggestion
      if (chart) {
        const datasets = chartSettings['dataset'].map(drupalData => {
          label: drupalData.chart.label,
          title: drupalData.chart.label,
          data: drupalData.plots,
          backgroundColor: drupalData.chart.color
        });
```